### PR TITLE
Mrc 1850: upgrade shiny server

### PR DESCRIPTION
**Pull request by richfitz**
This PR includes the tweaks used to update the shiny server, along with documentation about what I did.

Several apps were deleted - these are all out of date now:

* typochallenge: this is over and has been displaying a "this is over" page for ages now
* flupractical2: the deploy key has expired and is on a non-mrc-ide account so I can't fix it, but it was for a practical 2 years ago
* stochasticity-practical/*: these have rotted, but were for a practical 2 years ago and replaced by the msc-idm-20xx pages

### CODE REVIEW

File: upgrade.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+```
+./scripts/provision_all
+./scripts/sync_server
+```
+
+to update all apps. This will possibly work, but it's not guaranteed and will be a bit tedious to recover from. After a major upgrade I worked through the list of apps and categorised problems - some were due to missing libraries on the server due to the upstream docker image changing, some where deploy keys that had been removed.
+
+```
+./scripts/provision msc-idm-2019
+./scripts/provision msc-idm-2020
```
Comment (hillalex) [inline upgrade.md:57]: unclear what these lines represent - provisioning individual apps?  should there also be a `./scripts/sync_app app_name`?

File: upgrade.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+Remove old volumes
+
+```
+docker volume rm shiny_dide_applications shiny_dide_staging
+```
+
+## Apps that require special care and tending:
+
+* `msc-idm-2019`: `./helpers/reload-msc-idm-2019`
+* `msc-idm-2019`: `./helpers/reload-msc-idm-2020`
```
Comment (hillalex) [inline upgrade.md:91]: 2019 -> 2020 on this line

File: upgrade.md [hunk 3]
```diff
@@ -52,11 +52,20 @@ Theoretically we can do
...
 to update all apps. This will possibly work, but it's not guaranteed and will be a bit tedious to recover from. After a major upgrade I worked through the list of apps and categorised problems - some were due to missing libraries on the server due to the upstream docker image changing, some where deploy keys that had been removed.
 
+To provision an app, use the `./scripts/provision` script, such as
+
 ```
-./scripts/provision msc-idm-2019
 ./scripts/provision msc-idm-2020
 ```
 
+and then check the staging server, as http://129.31.x.y:3838/staging/msc-idm-2020/
```
Comment (hillalex) [inline upgrade.md:61]: So once we've checked the staging server, is there a further step to release the app?

Comment (richfitz) [reply upgrade.md:61]: Set up this way, this was running on my desktop, so there was no release. But this meant that I could test the system without having the server down all day. Once I'd iterated through a couple of times (build, deploy, build, deploy) and confirmed all applications worked, I then did a much quicker deploy on the real machine the normal way

Comment (hillalex) [reply upgrade.md:61]: Ah got you, this is just for testing somewhere, cool

**Pull request merged**